scripts/gen_datafile.py

```diff
@@ -0,0 +1,134 @@
+import math, random, string, os, time
+from dataclasses import dataclass
+from enum import auto, StrEnum
+from multiprocessing import Pool
+from typing import Iterator
+
+
+class ConfigNames(StrEnum):
+    KILO = auto()
+    PERCENT = auto()
+    FULL = auto()
+
+
+@dataclass
+class TestFileConfig:
+    total_rows: int
+    unique_locations: int
+    range: tuple[float, float]
+    configName: ConfigNames
+
+
+TEMP_RANGE = (-99.9, 99.9)
+
+CONFIGS: dict[ConfigNames, TestFileConfig] = {
+    ConfigNames.KILO: TestFileConfig(1_000, 50, TEMP_RANGE, ConfigNames.KILO),
+    ConfigNames.PERCENT: TestFileConfig(
+        10_000_000, 10_00, TEMP_RANGE, ConfigNames.PERCENT
+    ),
+    ConfigNames.FULL: TestFileConfig(
+        1_000_000_000, 10_000, TEMP_RANGE, ConfigNames.FULL
+    ),
+}
+
+
+class TestFileGen:
+    config: TestFileConfig
+    keys: list[str] = []
+    rows: list[str] = []
+    pageSize: int = 16_000
```

Comment:
Given testing, I saw that generally between 5000 and 17500 were the best results, but the peak seemed to come at 16000 per page, consistently hovering around 11 seconds to generate 100_000 rows.